R/formatter.r

```diff
@@ -588,18 +588,28 @@ ordinal_spanish <- function() {
   )
 }
 
-#' Parse a text label to produce expressions for plotmath.
+#' Parse a label to produce expressions for plotmath.
+#'
+#' @param text A character vector to parse. Other R objects will be coerced to
+#' character if possible
 #'
 #' @seealso [plotmath()]
-#' @return a function with single parameter x, a character vector, that
-#'    returns a list of expressions
+#' @return An expression object.
+#'
 #' @export
 #' @examples
-#' parse_format()(c("alpha", "beta", "gamma"))
-parse_format <- function() {
-  function(x) {
-    lapply(as.character(x), function(x) parse(text = x, srcfile = NULL))
+#' parse_format(c("alpha", "beta", "gamma"))
+#' parse_format(1:5)
+parse_format <- function(text) {
```

Comment:
You need to preserve the original structure of this function you code should look like:

```R
parse_format <- function() {
  function(x) {
     x <- as.character(x) 
     out <- ...
     ...
  }
}
```